### ISSUE TITLE
interleave_mode is not set correctly

### ISSUE THREAD

**Issue by Yuxiza** [coradoya/charls-rs#5]
charls need to set interleave_mode when more than one component in a frame
like this
```rust
let interleave_mode = match frame_info.component_count {
    1 => charls_interleave_mode_none,
    3 => charls_interleave_mode_sample,
    4 => charls_interleave_mode_line,
    _ => return Err(xxx),
};
```
Im not really sure about how to set this when having more than 4 components.

**Comment by dougyau** [comment]
Hi, from what I know, there should never be more than 3 components (Red Green and Blue) as Jpeg-LS does not support alpha channel. The interleave mode basically sets how the pixels will be encoded https://github.com/team-charls/charls/wiki/How-to-use-the-API#pixel-data-input-and-output-format.

**Comment by Yuxiza** [comment]
Thanks for your quick response!

> there should never be more than 3 components

In fact, Jpeg-LS CAN store more than 3 components according to the ITU-T T.87 manual which downloadlink can be found in jpeg.org. 

![Source image with multiple components](https://github.com/user-attachments/assets/5b7ddc99-6b13-4a7a-8821-4d295c8c178e)

And most of rgb image are encoded in sample_mode and image-rs also provide bytes in sample_mode (I believe). So, I think provide an api for users to chose interleave mode is necessary. Otherwise, users have to convert them manually.

**Comment by dougyau** [comment]
I have created the PR #6 exposing the set_interleave_mode of the CharLS codec. Can you check it? I think this should work for this use case.

**Comment by Yuxiza** [comment]
I think that's ok. Thanks!

**Comment by dougyau** [comment]
Released on [v0.4.0](https://github.com/coradoya/charls-rs/releases/tag/v0.4.0)
